game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -396,40 +396,40 @@ public void changeChanceDecrementOrIncrementOnSuccessOrFailure(final IDelegateBr
     }
   }
 
-  protected Map<String, AttachmentProperty<?>> createPropertyMap() {
-    return ImmutableMap.<String, AttachmentProperty<?>>builder()
+  protected Map<String, ModifiableProperty<?>> createPropertyMap() {
+    return ImmutableMap.<String, ModifiableProperty<?>>builder()
         .put("conditions",
-            AttachmentProperty.of(
+            ModifiableProperty.of(
                 this::setConditions,
                 this::setConditions,
                 this::getConditions,
                 this::resetConditions))
         .put("conditionType",
-            AttachmentProperty.of(
+            ModifiableProperty.of(
                 this::setConditionType,
                 this::setConditionType,
                 this::getConditionType,
                 this::resetConditionType))
         .put("invert",
-            AttachmentProperty.of(
+            ModifiableProperty.of(
                 this::setInvert,
```

Comment:
This line (and several others) are triggering null type safety warnings, probably due to the auto-unboxing.

Reply:
This is unfortunate, because internally the values are being stored as primitives 🤔

Reply:
Maybe there's some kind of utility wrapper we can add to automatically throw a checked exception when the input value is `null` before passing the (unboxed) value on to the actual setter?  Not sure if one utility would be required per primitive type.

Reply:
Here's a rough example of what I was thinking:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java b/game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java
index beb984dde..1aef9de8f 100644
--- a/game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java
+++ b/game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java
@@ -144,5 +144,19 @@ public interface ModifiableProperty<T> {
   @FunctionalInterface
   static interface ExceptionConsumer<T> {
     void accept(T object) throws Exception;
+
+    static ExceptionConsumer<Boolean> of(final ThrowingBooleanConsumer consumer) {
+      return value -> {
+        if (value == null) {
+          throw new Exception("value cannot be null"); // or whatever checked exception ExceptionConsumer ultimately throws
+        }
+        consumer.accept(value.booleanValue());
+      };
+    }
+  }
+
+  @FunctionalInterface
+  interface ThrowingBooleanConsumer {
+    void accept(boolean value) throws Exception;
   }
 }
diff --git a/game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java b/game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java
index a10b17634..8f69588cf 100644
--- a/game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java
+++ b/game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java
@@ -413,7 +413,7 @@ public abstract class AbstractConditionsAttachment extends DefaultAttachment imp
                 this::resetConditionType))
         .put("invert",
             ModifiableProperty.of(
-                this::setInvert,
+                ModifiableProperty.ExceptionConsumer.of(this::setInvert),
                 this::setInvert,
                 this::getInvert,
                 this::resetInvert))
```

Wrapper method names/location could use some refinement.

Reply:
Actually, these warnings were being raised because I had enabled annotation-based null analysis while investigating something for #2885.  Since we haven't decided to enable that yet, let's ignore it for now.